source/agora/node/GossipProtocol.d

```diff
@@ -80,6 +80,16 @@ public class GossipProtocol
             this.network.sendTransaction(tx);
             this.ledger.tryNominateTXSet();
         }
+
+        if (this.enroll_man.needRevealPreimage(this.ledger.getBlockHeight()))
```

Comment:
The `needRevealPreimage` check at that location causes unnecessary calls.
I think it would be better to check when the height of the block changes as a timing for checking `next_reveal_height`